Source/Core/DolphinWX/InputConfigDiag.cpp

```diff
@@ -447,15 +451,48 @@ void ControlDialog::AppendControl(wxCommandEvent& event)
   UpdateGUI();
 }
 
+void GamepadPage::EnableSettingControl(const std::string& group_name, const std::string& name,
+                                       const bool enabled)
+{
+  const auto box_iterator =
+      std::find_if(control_groups.begin(), control_groups.end(), [&group_name](const auto& box) {
+        return group_name == box->control_group->name;
+      });
+  if (box_iterator == control_groups.end())
+    return;
+
+  const auto* box = *box_iterator;
+  const auto it =
+      std::find_if(box->options.begin(), box->options.end(), [&name](const auto& pad_setting) {
+        return pad_setting->wxcontrol->GetLabelText() == name;
+      });
+  if (it == box->options.end())
+    return;
+  (*it)->wxcontrol->Enable(enabled);
+}
+
 void GamepadPage::AdjustSetting(wxCommandEvent& event)
 {
-  ((PadSetting*)((wxControl*)event.GetEventObject())->GetClientData())->UpdateValue();
+  const auto* const control = static_cast<wxControl*>(event.GetEventObject());
+  auto* const pad_setting = static_cast<PadSetting*>(control->GetClientData());
+  pad_setting->UpdateValue();
 }
 
-void GamepadPage::AdjustSettingUI(wxCommandEvent& event)
+void GamepadPage::AdjustBooleanSetting(wxCommandEvent& event)
 {
-  m_iterate = !m_iterate;
-  ((PadSetting*)((wxControl*)event.GetEventObject())->GetClientData())->UpdateValue();
+  const auto* const control = static_cast<wxControl*>(event.GetEventObject());
+  auto* const pad_setting = static_cast<PadSettingCheckBox*>(control->GetClientData());
+  pad_setting->UpdateValue();
+
+  // TODO: find a cleaner way to have actions depending on the setting
```

Comment:
I've thought about storing this sort of things in the setting itself, including conditions on which the setting should be enabled, etc.
or even a lamdba to have custom actions for settings.
But the issue is, this isn't going to be specific to wx, so I'm not sure what we could do in a lambda